controllers/image_generator.go

```diff
@@ -96,6 +97,12 @@ func ImageCreate(c client.Client) (bool, ctrl.Result) {
 			return false, ctrl.Result{Requeue: true}
 		}
 	}
+
+	if err := ig.validatePeerPodsConfigs(); err != nil {
+		igLogger.Info("error validating peer-pods configs", "err", err)
+		return false, ctrl.Result{Requeue: true, RequeueAfter: 30 * time.Second}
+	}
+
```

Comment:
IIUC it is assumed that if the peer pods config is valid at image creation time, it cannot be invalidated in any way and doesn't need to be validated again at image delete time. If yes, maybe make it clear in a comment or in the change log. Otherwise maybe move the check to `imageJobRunner()`. My motivation is that `imageJobRunner()` re-gets the CM from the server and we have no formal guarantee it wasn't changed by some external action.

Reply:
This is indeed assumed.
I added the validation also to deletion (originally i didn't add it as deletion requires less parameters in some cases but i guess it won't hurt to over-validate if it isn't expected to be changed and shouldn't harm functioning.. )

Reply:
Yes it isn't expected to change but since it is still technically possible, it seems better to make the code proof to that. Thanks !

Reply:
My actual concern is that the CM is retrieved from the server in `imageJobRunner()` and it looks like this is the place where we should ensure it is valid.

Reply:
Technically the validation is for the values used for the job itself (which mounts the secret&cm), and practically it's also validates the values used by the CAA which starts much later (this is why at first i did it in the main reconcile loop( and switched to avoid fetchCloudProvider code duplication))